src/kernel/src/shm/mod.rs

```diff
@@ -28,18 +28,11 @@ impl SharedMemoryManager {
         let flags: OpenFlags = i.args[1].try_into().unwrap();
         let mode: u32 = i.args[2].try_into().unwrap();
 
-        if (flags & OpenFlags::O_ACCMODE != OpenFlags::O_RDONLY)
-            || (flags & OpenFlags::O_ACCMODE != OpenFlags::O_RDWR)
-        {
+        if flags & OpenFlags::O_ACCMODE | OpenFlags::O_RDWR != OpenFlags::O_RDWR {
```

Comment:
This is quite confusing. Better to change to `(flags | OpenFlags::O_RDWR) & OpenFlags::O_ACCMODE != OpenFlags::O_RDWR`.